src/Symfony/Component/Console/Attribute/Input.php

```diff
@@ -0,0 +1,132 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Console\Attribute;
+
+use Symfony\Component\Console\Attribute\Reflection\ReflectionMember;
+use Symfony\Component\Console\Exception\LogicException;
+use Symfony\Component\Console\Input\InputInterface;
+
+#[\Attribute(\Attribute::TARGET_PARAMETER | \Attribute::TARGET_PROPERTY)]
+final class Input
+{
+    /**
+     * @var array<string, Argument>
+     */
+    private array $arguments = [];
+
+    /**
+     * @var array<string, Option>
+     */
+    private array $options = [];
+
+    /**
+     * @var array<string, self>
+     */
+    private array $inputs = [];
+
+    private \ReflectionClass $class;
+
+    public static function tryFrom(\ReflectionParameter|\ReflectionProperty $member): ?self
+    {
+        $reflection = new ReflectionMember($member);
+
+        if (!$self = $reflection->getAttribute(self::class)) {
+            return null;
+        }
+
+        $type = $reflection->getType();
+
+        if (!$type instanceof \ReflectionNamedType) {
+            throw new LogicException(\sprintf('The input %s "%s" must have a named type.', $reflection->getMemberName(), $member->name));
+        }
+
+        if (!class_exists($class = $type->getName())) {
+            throw new LogicException(\sprintf('The input class "%s" does not exist.', $type->getName()));
+        }
+
+        $self->class = new \ReflectionClass($class);
+
+        foreach ($self->class->getProperties() as $property) {
+            if (!$property->isPublic() || $property->isStatic()) {
+                continue;
+            }
+
+            if ($argument = Argument::tryFrom($property)) {
+                $self->arguments[$property->name] = $argument;
+                continue;
+            }
+
+            if ($option = Option::tryFrom($property)) {
+                $self->options[$property->name] = $option;
+                continue;
+            }
+
+            if ($input = static::tryFrom($property)) {
```

Comment:
The class is final, you can use `self`.
```suggestion
            if ($input = self::tryFrom($property)) {
```